src/github.com/matrix-org/dendrite/roomserver/storage/sql.go

```diff
@@ -240,6 +252,24 @@ func (s *statements) selectEventStateKeyNID(eventStateKey string) (eventStateKey
 	return
 }
 
+func (s *statements) selectEventStateKeyNIDs(eventStateKeys []string) ([]types.IDPair, error) {
+	rows, err := s.selectEventStateKeyNIDsStmt.Query(pq.StringArray(eventStateKeys))
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	results := make([]types.IDPair, len(eventStateKeys))
+	i := 0
```

Comment:
Why do you do this? In other projects, like [go-neb](https://github.com/matrix-org/go-neb/blob/master/src/github.com/matrix-org/go-neb/database/schema.go#L258), we just have a slice and call `append()`. I imagine this is being done for allocation reasons (once vs many with `append()` if it has to resize) but you don't comment that this is the case.